lsp/java/src/main/java/com/itsaky/lsp/java/actions/common/ImportOrdererAction.java

```diff
@@ -27,25 +34,57 @@ protected int getTitleTextRes() {
   }
 
   @Override
-  public Boolean execAction(ActionData data) {
+  public void prepare(ActionData data) {
+    super.prepare(data);
+    if (getVisible()) {
+      if (!hasRequiredData(data)) {
+        markInvisible();
+        return;
+      }
+
+      setVisible(true);
+      setEnabled(true);
+    }
+  }
+
+  @Override
+  public String execAction(ActionData data) {
+    final long start = System.currentTimeMillis();
     try {
       CodeEditor editor = requireEditor(data);
-      String output = ImportOrderer.reorderImports(editor.getText().toString(), JavaFormatterOptions.Style.GOOGLE);
-      editor.setText(output);
-      return true;
-    } catch (FormatterException fe) {
-      fe.printStackTrace();
-      return false;
+      Content content = editor.getText();
+      if (content != null) {
+
+        String output =
+            ImportOrderer.reorderImports(content.toString(), JavaFormatterOptions.Style.GOOGLE);
+        log.info("Reorder imports in", System.currentTimeMillis() - start + "ms");
+        return output;
+      }
+
+    } catch (FormatterException e) {
+      log.error("Failed to reorder imports", e);
+      return null;
     }
+    return null;
   }
 
   @Override
   public boolean getRequiresUIThread() {
-    return true;
+    return false;
   }
 
   @Override
   public void setLabel(String arg0) {}
 
+  @Override
+  public void postExec(ActionData data, Object result) {
+    super.postExec(data, result);
+    String text = (String) result;
```

Comment:
This will throw `ClassCastException` if `result` is not a string (which will definitely happen if there were any errors faced in `execAction`). First, check if `result` is a `String` or not. If it is, then cast it to `String` and set to editor, else, log an error message.